controllers/system/system_controller.go

```diff
@@ -1374,6 +1374,17 @@ func (r *SystemReconciler) ReconcileResource(client *gophercloud.ServiceClient,
 		return err
 	}
 
+	certList := make([]certificates.Certificate, 0)
+	for _, cert := range systemInfo.Certificates {
+		for _, instCert := range *instance.Spec.Certificates {
+			if cert.Signature == instCert.Signature {
+				certList = append(certList, cert)
+				break
+			}
+		}
+	}
+	systemInfo.Certificates = certList
```

Comment:
the update of the systemInfo should stay in systemInfo.PopulateSystemInfo

Reply:
The parameters of PopulateSystemInfo doesnt have the instance in it
WHich is restricting me to update the PopulateSystemInfo